jetty-util/src/main/java/org/eclipse/jetty/util/Fields.java

```diff
@@ -253,7 +253,9 @@ private Field(String name, List<String> values, String... moreValues)
 
         public boolean equals(Field that, boolean caseSensitive)
         {
-            if (this == that)
+            @SuppressWarnings("ReferenceEquality")
+            boolean isThisInstance = (this == that);
```

Comment:
I don't think this is needed at all. It is a common idiom to implement `equals()` by first testing for reference equality - we don't want to clutter all `equals()` implementations with these extra 2 lines.